container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -82,6 +82,52 @@ def check_deployment_status(deployment):
     return deployment_status
 
 
+def gitops_checkout_tempdir(git_url, git_dir, git_branch,
+                            appname_dir, deploy_type):
+    """Method to checkout gitops repository based on git_branch
+        and push the Compiled Gateway Configs(all.yaml)
+        file to Gitopsrepo.
+    """
+    # Create temporary dir
+    tempdir = tempfile.mkdtemp()
```

Comment:
It is preferable to use the context manager here (the with construct as in the example below)
https://docs.python.org/3/library/tempfile.html#examples

The advantage is, that the context manager will be more fault tolerant, for example it will make sure that tempdir is deleted even if an error occurs while running the code.

It is not critical, but normally it is preferable.